tests/test_testRepo.py

```diff
@@ -126,15 +126,16 @@ def testAddDataIdValue(self):
         with self.assertRaises(ValueError):
             addDataIdValue(self.butler, "detector", 101, nonsenseField="string")
 
-        # Keywords imply different instruments
-        with self.assertRaises(RuntimeError):
-            addDataIdValue(self.butler, "exposure", 101, instrument="dummyCam", physical_filter="k2020")
```

Comment:
I don't understand why this test case has been removed. The extra dimensions are inconsistent (`physical_filter` implies `"notACam"`), so it should still fail.

Reply:
I played around with a number of adjustments to try to make this work, and eventually I convinced myself that if the system is allowed to create new dimension values, there's no way to distinguish between this case and missing implied dimension records more generally, at least not without the kind of exhaustive search that the butler deliberately disables because in real-world contexts it always indicates a mistake (if someone queries for data IDs with `detector=4` in a real repo without specifying the instrument, they're not asking for to cartesian-product iterate over all instruments that have a detector 4).

Switching to only allowing implied dimension records to be created might mitigate this somewhat, but it wouldn't save this test (that's one of the things I tried), because `physical_filter` is an implied dimension of `exposure`, so creating one that matches the undesired `instrument` with the given filter name would still be permitted.

So I think the choice is between this behavior and reverting back to the original version of this branch, where downstream tests were still required to provide implied dimension value explicitly.

Reply:
Ah!  Nevermind - the thread just below this one made me realize the distinction you want made: you do not want records created for any dimension that is referenced explicitly in the call to `addDataIdValue`; those are asserted to already exist, and only those not mentioned should be created automatically.  It seems obvious now; I was too deep in the weeds (having basically dropped the original set of dimensions given on the floor in favor of its expansion) to see it before.  I'll be back with another of this shortly.

Reply:
Ok, done!  This let me restore the original test to exactly what it was before.